src/v/cluster/tx_coordinator_mapper.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "cluster/metadata_cache.h"
+#include "cluster/partition_manager.h"
+#include "cluster/tm_tx_hash_ranges.h"
+#include "hashing/murmur.h"
+#include "kafka/protocol/types.h"
+#include "kafka/types.h"
+#include "model/fundamental.h"
+#include "model/namespace.h"
+#include "seastar/core/sharded.hh"
+#include "seastarx.h"
+
+#include <seastar/core/reactor.hh>
+
+namespace cluster {
+
+inline model::partition_id get_partition_from_default_distribution(
+  uint32_t tx_id_hash, int32_t partitions_amount) {
```

Comment:
would we benefit from using a `named_type` for the hash?

Reply:
Maybe not named_type but just using. I need to save arithmetic operations

Reply:
fwiw named_type lets you use arithmetic ops... or you can access the underlying value directly with operator()